JSDOMParser.js

```diff
@@ -60,13 +60,23 @@
       .replace(/&(quot|amp|apos|lt|gt);/g, function (match, tag) {
         return entityTable[tag];
       })
-      .replace(
-        /&#(?:x([0-9a-z]{1,4})|([0-9]{1,4}));/gi,
-        function (match, hex, numStr) {
-          var num = parseInt(hex || numStr, hex ? 16 : 10); // read num
-          return String.fromCharCode(num);
+      .replace(/&#(?:x([0-9a-f]+)|([0-9]+));/gi, function (match, hex, numStr) {
+        var num = parseInt(hex || numStr, hex ? 16 : 10);
+
+        // these character references are replaced by a conforming HTML parser
+        if (num == 0 || num > 0x10ffff || (num >= 0xd800 && num <= 0xdfff)) {
+          num = 0xfffd;
+        }
+
+        // code points beyond the BMP must be converted to a surrogate pair when using String.fromCharCode
+        if (num > 0xffff) {
+          var high = 0xd800 + (((num - 0x10000) & 0xffc00) >> 10);
+          var low = 0xdc00 + ((num - 0x10000) & 0x3ff);
+          return String.fromCharCode(high) + String.fromCharCode(low);
         }
```

Comment:
Could we use `String.fromCodePoint` directly here? It looks like that is supported from NodeJS 4.0 and we require a higher nodeJS version so we should be OK. That would save me (and any future readers) from having to reason about the hex code. 😅

Reply:
Yes, `fromCodePoint` would obviate a nice chunk of code. I was unaware of the baseline and a lot of code presumably predates the policy, so I jump through the hoops. 

I'll make the change when I have a minute.

Reply:
Ah, great - I may also be able to do it directly - I didn't want to do so without asking just in case there was a reason not to that I was missing.

Reply:
Done.